### DISCUSSION TITLE
Can you live without tables?

### DISCUSSION THREAD

**Discussion by derWalter** [Zettlr/Zettlr#4341] [General]
Hey guys, I have decided to quit writing in Libre Office Writer and I am moving all my thesis projects over to Zettlr.
Right now I export to word (I would love to use ICML but it is a mess with around 30 abitrary styles exported, lists broken and so forth)
because I am quite fluid with InDesign and I have no clue about LaTex,
so I am totally handicapped regarding the final optics of my pdf output.
So thats the workflow I am working on right now.

But, for the love of god, I cant work without tables :D
- a thing I love to do from time to time is put two lists next to each other.
Usually I use a table with one row and two cells, where I put my lists in.
- Or if I want to write in two columns just for a segment, I do the same.

But I cant get it (linebreaks or lists in cells) to work in zettlr, I guess
handling tables is even worse for me in beta 3.0 :P

The whole situation reminds me of mid 90ies when tables where introduced
into html and it took years for the browsers to get them displayed in a consistent way,
which I never managed to understand WHY they had so many problems with it ^^

### So how are you other users are all dealing with this? 
Did you just unlearn tables and live without em, or
do you maybe know any easy hacks I haven't stumbled upon in my research?
I have read some people do the tables in other Software like Writer and paste a screenshot
into Zettlr ... lol

### and a question the devs, or anyone else ;)
would there be ANY problem in using 100% html tables? So that the html code for the table,
is rendered and therefore invisible to the user (like it is right now) but with all the features
html provides? So it looks and feels like in a wysiwyg word processor... tables offer so much features,
that its apita to write them in html syntax, before I have to manually deal with `<ul>` and co, I just have to let it be I guess.

**Comment by nathanlesage** [comment]
> would there be ANY problem in using 100% html tables?

Yes, it's insanely difficult to implement. I do still have a full rework of the TableEditor in mind, but that has to currently stay behind as other issues are more pressing. Tables are indeed very difficult to properly do, I had to figure out over the past six years of developing that app …

My best advice to workaround that right now would be to use HTML if you need some complexity or try to see if you can automatically generate the tables (mainly important for data science work where you can have an R or Python script automatically generate table code into another file that you can then include in your document at the appropriate position). I know it's not the most beautiful method, but my time unfortunately is also limited and there's only so much I can tend to at once :/

**Reply by derWalter** [reply]
Oh, I see, the rendering is so difficult... maybe there is an open sourced engine which could be used to render just this part and it only gets refreshed once it changes? ... but I guess u already checked that out.

yeah, sadly, one cant apply at Arbeitsamt for being an open source developer and get hartz IV plus something for every hour of develpment :| would serve the world well :P

so I should just use some html table generator and drop in the code and work from there .... mhhhh could we at least get a html table generator in the meantime, so I dont need a bookmark in the browser/second app, just for generating the html out of zettlr? that would already be helpful (for people who know how to handle html)

**Comment by exanrauzer** [comment]
Go to GPT to generate the table in markdown;)
